arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -299,15 +323,20 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
 
       if (tmp.ok()) {
         result = DBServerAgencySyncResult(
-            true,
-            report.hasKey("Plan") ? report.get("Plan").get("Version").getNumber<uint64_t>() : 0,
-            report.hasKey("Current")
-                ? report.get("Current").get("Version").getNumber<uint64_t>()
-                : 0);
+          true,
+          report.hasKey("Plan") ?
+            report.get("Plan").get("Version").getNumber<uint64_t>() : 0,
+          report.hasKey("Current") ?
+            report.get("Current").get("Version").getNumber<uint64_t>() : 0,
+          report.hasKey("Plan") ?
+            report.get("Plan").get("Index").getNumber<uint64_t>() : 0,
+          report.hasKey("Current") ?
+            report.get("Current").get("Index").getNumber<uint64_t>() : 0);
```

Comment:
This will do up to 4 lookups in the object for `Plan`, and up to 4 for `Current`.
It would be better to do it like this:
```
VPackSlice planSlice = report.get("Plan");
VPackSlice currentSlice = report.get("Current");
...
```suggestion
          planSlice.isObject() ?
            planSlice.get("Version").getNumber<uint64_t>() : 0,
          currentSlice.isObject() ?
            currentSlice.get("Version").getNumber<uint64_t>() : 0,
          planSlice.isObject() ?
            planSlice.get("Index").getNumber<uint64_t>() : 0,
          currentSlice.isObject() ?
            currentSlice.get("Index").getNumber<uint64_t>() : 0);
```